pkg/master/controller.go

```diff
@@ -155,15 +174,14 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
 			Labels:    map[string]string{"provider": "kubernetes", "component": "apiserver"},
 		},
 		Spec: api.ServiceSpec{
-			Ports: []api.ServicePort{{Port: servicePort, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(servicePort)}},
+			Ports: ports,
 			// maintained by this code, not by the pod selector
 			Selector:        nil,
 			ClusterIP:       serviceIP.String(),
```

Comment:
I guess I'm OK with adding NodePort as an option (it would be nice to add
LoadBalancer, too). But you have to put all the endpoints stuff back the
way it was. If you want to change that, do it in a separate PR, it's a
totally orthogonal problem.

On Fri, Sep 25, 2015 at 10:38 AM, jay vyas <notifications@github.com> wrote:

> In pkg/master/controller.go
> <https://github.com/kubernetes/kubernetes/pull/13978#discussion_r40455228>
> :
>
> > @@ -155,15 +174,14 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
> >  			Labels:    map[string]string{"provider": "kubernetes", "component": "apiserver"},
> >  		},
> >  		Spec: api.ServiceSpec{
> > -			Ports: []api.ServicePort{{Port: servicePort, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(servicePort)}},
> > +			Ports: ports,
> >  			// maintained by this code, not by the pod selector
> >  			Selector:        nil,
> >  			ClusterIP:       serviceIP.String(),
>
> Yup I think we are all in agreement ... The only question is :
> 1. Does this feature make sense as a standalone addition to the toolbox of
> options available for kube deployments ?
>
> 2. I think "yes" (correct me if wrong)... But I think @thockin
> <https://github.com/thockin> is suggesting at the very-least, that I
> *clarify* the context behind the hacky "launch kubelets via nodeport
> apiserver" example, which can confuse people into thinking that ClusterIP
> is not sufficient for internal loadbalanced access.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13978/files#r40455228>.
>